homeassistant/components/motioneye/camera.py

```diff
@@ -40,18 +51,48 @@
     listen_for_new_cameras,
 )
 from .const import (
+    CONF_ACTION,
     CONF_CLIENT,
     CONF_COORDINATOR,
     CONF_STREAM_URL_TEMPLATE,
     CONF_SURVEILLANCE_PASSWORD,
     CONF_SURVEILLANCE_USERNAME,
     DOMAIN,
     MOTIONEYE_MANUFACTURER,
+    SERVICE_ACTION,
+    SERVICE_SET_TEXT_OVERLAY,
+    SERVICE_SNAPSHOT,
     TYPE_MOTIONEYE_MJPEG_CAMERA,
 )
 
 PLATFORMS = ["camera"]
 
+SCHEMA_TEXT_OVERLAY = vol.In(
+    [
+        KEY_TEXT_OVERLAY_DISABLED,
+        KEY_TEXT_OVERLAY_TIMESTAMP,
+        KEY_TEXT_OVERLAY_CUSTOM_TEXT,
+        KEY_TEXT_OVERLAY_CAMERA_NAME,
+    ]
+)
+SCHEMA_SERVICE_SET_TEXT = vol.Schema(
+    vol.All(
+        {
+            vol.Optional(KEY_TEXT_OVERLAY_LEFT): SCHEMA_TEXT_OVERLAY,
+            vol.Optional(KEY_TEXT_OVERLAY_CUSTOM_TEXT_LEFT): cv.string,
+            vol.Optional(KEY_TEXT_OVERLAY_RIGHT): SCHEMA_TEXT_OVERLAY,
+            vol.Optional(KEY_TEXT_OVERLAY_CUSTOM_TEXT_RIGHT): cv.string,
+        },
+        cv.has_at_least_one_key(
+            KEY_TEXT_OVERLAY_LEFT,
+            KEY_TEXT_OVERLAY_CUSTOM_TEXT_LEFT,
+            KEY_TEXT_OVERLAY_RIGHT,
+            KEY_TEXT_OVERLAY_CUSTOM_TEXT_RIGHT,
+        ),
+    ),
+    extra=vol.ALLOW_EXTRA,
```

Comment:
Allowing extra is incorrect. It was probably added to not error when passing entity id.